qiskit/aqua/operators/state_functions/dict_state_fn.py

```diff
@@ -0,0 +1,303 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" An Object to represent State Functions constructed from Operators """
+
+from typing import Union
+import itertools
+import numpy as np
+from scipy import sparse
+
+from qiskit.result import Result
+from qiskit.circuit import ParameterExpression
+
+from ..operator_base import OperatorBase
+from . import StateFn
+from ..combo_operators import ListOp
+
+
+class DictStateFn(StateFn):
+    """ A class for representing state functions and measurements.
+
+    State functions are defined to be complex functions over a single binary string
+    (as compared to an operator,
+    which is defined as a function over two binary strings, or a function taking a binary
+    function to another
+    binary function). This function may be called by the eval() method.
+
+    Measurements are defined to be functionals over StateFns, taking them to real values.
+    Generally, this real value
+    is interpreted to represent the probability of some classical state (binary string)
+    being observed from a
+    probabilistic or quantum system represented by a StateFn. This leads to the
+    equivalent definition, which is that
+    a measurement m is a function over binary strings producing StateFns, such that
+    the probability of measuring
+    a given binary string b from a system with StateFn f is equal to the inner product
+    between f and m(b).
+
+    NOTE: State functions here are not restricted to wave functions, as there is
+    no requirement of normalization.
+    """
+
+    # TODO maybe break up into different classes for different fn definition primitives
+    # TODO allow normalization somehow?
+    def __init__(self,
+                 primitive: Union[str, dict, Result] = None,
+                 coeff: Union[int, float, complex, ParameterExpression] = 1.0,
+                 is_measurement: bool = False) -> None:
+        """
+            Args:
+                primitive: The operator primitive being wrapped.
+                coeff: A coefficient by which to multiply the state function.
+                is_measurement: Whether the StateFn is a measurement operator.
```

Comment:
This refers to some StateFn which is what - this class being a subclass of it? Since its not constructed yet it cannot be anything. Is this more like use_measurement where you are asking this class to use measurements in what it does. The is_measurement is confusing as normally that type of language would be a method to return whether a property was true/false, in this case measurement